code/datums/components/caltrop.dm

```diff
@@ -0,0 +1,49 @@
+/datum/component/caltrop
+	var/damage
+	var/flags
+
+	var/cooldown = 0
+
+/datum/component/caltrop/Initialize(_damage, _flags = NONE)
+	RegisterSignal(list(COMSIG_MOVABLE_CROSSED), .proc/Crossed)
+
+/datum/component/caltrop/proc/Crossed(atom/movable/AM)
+	var/atom/A = parent
+	if(!A.has_gravity())
+		return
+
+	if(ishuman(AM))
+		var/mob/living/carbon/human/H = AM
+		if(PIERCEIMMUNE in H.dna.species.species_traits)
+			return
+
+		if((flags & CALTROP_IGNORE_WALKERS) && H.m_intent == MOVE_INTENT_WALK)
+			return
+
+		var/picked_def_zone = pick("l_leg", "r_leg")
```

Comment:
Think we have defines for these? not sure though

Reply:
I couldn't find any defines for them.